src/Catalog/Monitoring/Model/PackageTimestampMetadata.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Providers
+{
+    /// <summary>
+    /// Represents the timestamp metadata for a single package in a package source.
+    /// </summary>
+    public class PackageTimestampMetadata : INuGetResource
+    {
+        public PackageTimestampMetadata()
+        {
+            Exists = false;
+            Created = null;
+            LastEdited = null;
+            Deleted = null;
+        }
+
+        public PackageTimestampMetadata(DateTime? deleted)
```

Comment:
>PackageTimestampMetadata [](start = 15, length = 24)

This ctor and the next are pretty similar when inspecting the types of parameters. Better to have static CreateDeleted and CreateExisting methods that describe the type of record you are initializing